### ISSUE TITLE
Ignored directives report errors in disabled regions

### ISSUE THREAD

**Issue by jjonescz** [dotnet/roslyn#78157]
**Version Used**: main

**Steps to Reproduce**:

```cs
#if false

System.Console.WriteLine("""
    #:sdk test
    """);

#endif
```

**Expected Behavior**: No error reported.

**Actual Behavior**:

```cs
// /Program.cs(4,6): error CS9298: '#:' directives can be only used in file-based programs ('-features:FileBasedProgram')
//     #:sdk test
Diagnostic(ErrorCode.ERR_PPIgnoredNeedsFileBasedProgram, ":").WithLocation(4, 6),
// /Program.cs(4,6): error CS9299: '#:' directives cannot be after '#if' directive
//     #:sdk test
Diagnostic(ErrorCode.ERR_PPIgnoredFollowsIf, ":").WithLocation(4, 6)
```

**Comment by RikkiGibson** [comment]
What is the representation of the content in the disabled region? I thought there would at least be a "string literal token" which contains the `#:`, and we would understand that it should not be lexed as a directive.

**Comment by CyrusNajmabadi** [comment]
This is by design.  Our preprocessor is not an actual *pre* processor.  It's an in-situ lexer-controlling system.  So once in a disabled block, there is no actual code, just scanning line for other directives.  :)

**Comment by jjonescz** [comment]
Still as the linked PR shows, we could improve the UX for example at least for `#:` directives - by not reporting some errors in the disabled regions.

**Comment by CyrusNajmabadi** [comment]
Sure. It's up to you as to the semantics you want for that feature :)
